### PR TITLE
Continue sync daemon implementation

### PR DESCRIPTION
**Pull request by kislyuk**

### CODE REVIEW

File: daemons/dss-sync/app.py [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
+
 app = domovoi.Domovoi()
 
 s3_bucket = os.environ.get("DSS_S3_TEST_BUCKET")
 
 @app.s3_event_handler(bucket=s3_bucket, events=["s3:ObjectCreated:*"])
 def process_new_syncable_object(event, context):
-    bucket = boto3.resource("s3").Bucket(event['Records'][0]["s3"]["bucket"]["name"])
+    s3 = boto3.resource("s3")
+    bucket = s3.Bucket(event['Records'][0]["s3"]["bucket"]["name"])
```
Comment (ttung) [inline daemons/dss-sync/app.py]: I'm unfamiliar with S3 events, but could `['Records']` have more than one entry?

File: dss/__init__.py [hunk 2]
```diff
@@ -4,13 +4,18 @@
...
 import flask
 import connexion
 from connexion.resolver import RestyResolver
 from flask_failsafe import failsafe
 
 def get_logger():
-    return flask.current_app.logger
+    try:
+        return flask.current_app.logger
+    except RuntimeError:
```
Comment (ttung) [inline dss/__init__.py]: Is `RuntimeError` what's really thrown?

File: dss/events/handlers/sync.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import os, sys, datetime, json, time, hashlib
```
Comment (ttung) [inline dss/events/handlers/sync.py]: I'm not a huge fan of the multi-module import...

File: dss/events/handlers/sync.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+from contextlib import closing
+
+import boto3
+import botocore.session
+import urllib3
```
Comment (ttung) [inline dss/events/handlers/sync.py]: order your imports!

File: dss/events/handlers/sync.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+class GCStorageTransferClient(ClientWithProject):
+    SCOPE = ["https://www.googleapis.com/auth/cloud-platform"]
+
+class GCStorageTransferConnection(JSONConnection):
+    API_BASE_URL = "https://storagetransfer.googleapis.com"
+    API_VERSION = "v1"
+    API_URL_TEMPLATE = "{api_base_url}/{api_version}{path}"
+
+# TODO: access keys used here should be separate role credentials with need-based access
```
Comment (ttung) [inline dss/events/handlers/sync.py]: Should associate this `TODO` with someone.

File: dss/events/handlers/sync.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+            "status": "ENABLED",
+            "projectId": gcs.project,
+            "schedule": {
+                "scheduleStartDate": schedule_struct,
+                "scheduleEndDate": schedule_struct,
+                "startTimeOfDay": dict(hours=schedule_at.hour, minutes=schedule_at.minute)
+            },
+            "transferSpec": {
+                "awsS3DataSource": {
+                    "bucketName": os.environ["DSS_S3_TEST_BUCKET"],
```
Comment (ttung) [inline dss/events/handlers/sync.py]: this is repeated. want to extract a var for it?

File: daemons/dss-sync/app.py (hunk 1 above)
Comment (kislyuk) [inline daemons/dss-sync/app.py]: The docs at http://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html don't explicitly state it, but from what I know we can expect to receive no more than one record at a time.

File: dss/__init__.py (hunk 2 above)
Comment (kislyuk) [inline dss/__init__.py]: Yes:
```
Traceback (most recent call last):
  File "./tests/test.py", line 18, in <module>
    dss.get_logger()
  File "/Users/akislyuk/projects/data-store/dss/__init__.py", line 16, in get_logger
    return flask.current_app.logger
  File "/Users/akislyuk/projects/data-store/v/lib/python3.6/site-packages/werkzeug/local.py", line 347, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/Users/akislyuk/projects/data-store/v/lib/python3.6/site-packages/werkzeug/local.py", line 306, in _get_current_object
    return self.__local()
  File "/Users/akislyuk/projects/data-store/v/lib/python3.6/site-packages/flask/globals.py", line 51, in _find_app
    raise RuntimeError(_app_ctx_err_msg)
RuntimeError: Working outside of application context.

This typically means that you attempted to use functionality that needed
to interface with the current application object in a way.  To solve
this set up an application context with app.app_context().  See the
documentation for more information.
```

File: dss/events/handlers/sync.py (hunk 3 above)
Comment (kislyuk) [inline dss/events/handlers/sync.py]: Fixed.

File: dss/events/handlers/sync.py (hunk 4 above)
Comment (kislyuk) [inline dss/events/handlers/sync.py]: Can you clarify what you mean here? It's good practice to order imports into 3 groups: stdlib, third-party, and local library/relative. Within each group, do you want them ordered alphabetically?

File: dss/events/handlers/sync.py (hunk 5 above)
Comment (kislyuk) [inline dss/events/handlers/sync.py]: Done.

File: dss/events/handlers/sync.py (hunk 6 above)
Comment (kislyuk) [inline dss/events/handlers/sync.py]: Good point, done here and in tests.

File: dss/__init__.py (hunk 2 above)
Comment (ttung) [inline dss/__init__.py]: Wow.

**Comment by ttung** [comment]
This doesn't show me as a reviewer, but approved!

**Pull request merged**